### PR TITLE
fix oai ia imports

### PR DESCRIPTION
**Pull request by summer-cook**
# Coauthor
@aprilrieger

# Summary
imports with the oai ia parser were failing. This PR upgrades some of the overridden bulkrax files to fix these imports. 

# Related
https://github.com/scientist-softserv/atla_digital_library/issues/248

# Screenshot
can now import successfully with the OaiIA parser: 
image

# Acceptance Criteria
- [ ] as a user, I can import an oai feed with the oai ia parser

# Testing Instructions
1. Sign in and go to the /importers page
2. Click the button to create a new importer
3. Use the following applicable info to fill out the importer:

name: `AMBS -  Biblioteca Digital Anabautista`
admin_set_id: `admin_set/default`
limit: `25` <---- set this or you will get ALOT of entries
parser: `OAI - Internet Archive`
base_url: `https://archive.org/services/oai.php`
metadata_prefix: `oai_dc`
set: `collection:bibliotecadigitalanabautista`
collection_title: `AMBS -  Biblioteca Digital Anabautista`
institution_name: `Anabaptist Mennonite Biblical Seminary`
thumbnail_url: `https://archive.org/services/img/<%= identifier.split(':').last %>"}`

4. Ensure the importer says "completed" and works import successfully, report any errors you see with screenshots and behavior description, try to include as much detail as possible so we can pickup where you left off if it gets kicked back :)

### CODE REVIEW

File: app/parsers/bulkrax/application_parser_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# OVERRIDE Bulkrax 4.4.1 to get a patch for create_objects
+
+module Bulkrax
+  module ApplicationParser
+    module ApplicationParserDecorator
+      # override create_objects to get the patch from https://github.com/samvera-labs/bulkrax/pull/663
+      def create_objects(types = [])
+        types.each do |object_type|
```
Comment (bkiahstroud) [inline app/parsers/bulkrax/application_parser_decorator.rb]: 1. Without the `DEFAULT_OBJECT_TYPES` variable added by https://github.com/samvera-labs/bulkrax/pull/663/files, what does `types` end up being here? 
2. Also, can we please add a `TODO` comment that says to delete this decorator when upgrading Bulkrax to v5.x.x?

**Review by bkiahstroud — CHANGES_REQUESTED** [review]

File: app/parsers/bulkrax/application_parser_decorator.rb (hunk 1 above)
Comment (summer-cook) [reply app/parsers/bulkrax/application_parser_decorator.rb]: I realized i just got the PR in the comment wrong. PR #663 IS already in version 4.4.1, its this patch PR https://github.com/samvera-labs/bulkrax/pull/713 that we needed to pull in.

Comment (summer-cook) [reply app/parsers/bulkrax/application_parser_decorator.rb]: updating comments now though to also add the todo and fix the PR that is linked there!

Comment (summer-cook) [reply app/parsers/bulkrax/application_parser_decorator.rb]: @bkiahstroud done!!

**Comment by summer-cook** [comment]
@bkiahstroud i updated this branch with your comments and also realized i forgot to remove the overridden controller, so i added another commit.

reason for removal: all the overrides were no longer needed & had been either updated or added in in the version of bulkrax we upgraded to. so its better now to just have that file inherit.

**Comment by bkiahstroud** [comment]
@summer-cook @aprilrieger I see this was recently marked as a draft, is it ready for re-review?

**Comment by summer-cook** [comment]
> @summer-cook @aprilrieger I see this was recently marked as a draft, is it ready for re-review?

not yet, just trying to figure out a few more things

File: config/initializers/bulkrax.rb [hunk 2]
```diff
@@ -14,7 +14,7 @@
   config.default_work_type = 'Work'
 
   # Field to use during import to identify if the Work or Collection already exists.
-  # Default is 'source'.
+  # Default is 'work_identifier' which is 'source'
```
Comment (alishaevn) [inline config/initializers/bulkrax.rb]:
```suggestion
  # Default is 'source'; defined in #work_identifier.
```

**Review by alishaevn — CHANGES_REQUESTED** [review]

**Comment by summer-cook** [comment]
@bkiahstroud ok, NOW it is ready

**Review by alishaevn — APPROVED** [review]

File: app/parsers/bulkrax/oai_ia_parser.rb [hunk 3]
```diff
@@ -1,3 +1,6 @@
+# OVERRIDE: Bulkrax 4.4.1 to add collection title, noted below.
+# OVERRIDE: Bulkrax 4.4.1 to get a patch from Bulkrax 5.x.x for create_objects, noted below.
+# TODO: remove create_objects method when upgrading to Bulkrax 5.x.x
```
Comment (bkiahstroud) [inline app/parsers/bulkrax/oai_ia_parser.rb]: @summer-cook these comments reference `#create_objects`, but I don't see any overrides for that method anymore. Does that mean these comments can be removed? Or do they need to be updated to refer to `#create_collections` now?

Comment (summer-cook) [reply app/parsers/bulkrax/oai_ia_parser.rb]: @bkiahstroud yes - just removed those comments!

**Review by bkiahstroud — APPROVED** [review]

**Pull request merged**